assignment/impl/src/java/org/sakaiproject/assignment/impl/persistence/AssignmentRepositoryImpl.java

```diff
@@ -95,6 +96,28 @@ public void newAssignment(Assignment assignment) {
         }
     }
 
+    @Override
+    @Transactional
+    public void newAssignmentTimeSheet(AssignmentTimeSheet timeSheet) {
+        if (!existsAssignmentTimeSheet(timeSheet.getId())) {
+            sessionFactory.getCurrentSession().persist(timeSheet);
+        }
+    }
+
+    @Override
+    @Transactional
+    public void deleteAssignmentTimeSheet(AssignmentTimeSheet timeSheet) {
+        if(timeSheet!= null) {
+        	sessionFactory.getCurrentSession().delete(sessionFactory.getCurrentSession().merge(timeSheet));
```

Comment:
make sessionFactory.getCurrentSession() a local var since it's used more than once

Reply:
Moved to class declaration and updated every instance.